.github/workflows/ci.yml

```diff
@@ -0,0 +1,68 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: testsuite
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy: { matrix: { node_version: [10.x] } }
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Node v${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: npm-${{ hashFiles('package-lock.json') }}
+          restore-keys: npm-
+      - run: npm ci
+      - run: cp config.example.yml config.yml
+      - run: npm run lint
```

Comment:
Is it worth it to have a standalone step for linting? I have a feeling that we'll spend more time on installing a specific version of Node.js and installing the dependencies, even with caching.

Reply:
Was planning to also add 'depcheck' here to run at this step to ensure we don't have a bunch of unused packages.

Personally, I prefer different status reports for different things, my issue with the current implementation being that if you have issues in linting and the e2es, you don't notice the latter until the former has been fixed.

But yeah I need to figure out how to bring down the `npm ci` step down more generally, possiblly caching the build of node-sass or using the javascript implementation of node-sass, not the c one - the performance difference will be negligible for such a small project.

If I can't get the install/cache steps down to sub 1m, then agreed seems worth merging at least unit and lint.

Finding time to work on this again this week.